screens/Discover.js

```diff
@@ -41,14 +41,18 @@ class Discover extends Component {
       this.setState({ dataReceived: true })
     });
   }
+  _onWorkoutSelect(workout) {
+    this.props.navigation.navigate('WorkoutDetails', { workout: workout, discoverWorkout: true });
+  }
   render() {
     var discoverWorkoutViews;
     if (this.workouts.length > 0) {
```

Comment:
Any advantages of using for loop instead of `map` workouts object?

Reply:
Accessing some elements from `map` not working (will result in accessing `undefined` element) without bind(this), so to avoid that you need to use a for loop

Reply:
But what's the problem with binding? That's just one line of code..?

Reply:
Just extra work for the same result. Replaced.

Reply:
I meant mapping directly inside the return statement.

Reply:
On what line? Didn't understand what you meant. 😅

Reply:
I meant removing `for` loop, and map `workouts` object inside the return statement like you [do for any kind of lists in React](https://reactjs.org/docs/lists-and-keys.html#embedding-map-in-jsx).

Reply:
Though I need to have different results if workouts are loaded (n>0) or not loaded. And then my return contains many different elements unrelated to workout list: 
<img width="440" alt="Screen Shot 2019-07-07 at 20 45 00" src="https://user-images.githubusercontent.com/8983928/60776181-1d92bc00-a0f8-11e9-8ca1-9aebcda19f56.png">

Reply:
Wouldn't it work:
```jsx
{this.workouts.length > 0 && this.workouts.map((workout, index) =>  
                return <DiscoverItem workout={workout} key={index} onPress={(workout) => { this._onWorkoutSelect(workout) }} style={style} />
)}
{this.workouts.length === 0 && <Text>Loading</Text>} 
```

Reply:
I like it better when it is outside of the return statement, so return is smaller and cleaner itself. 
Text will be replaced with a spinner and maybe something else (animation or whatever)